cas-server-core/src/main/java/org/jasig/cas/ticket/registry/DefaultTicketRegistry.java

```diff
@@ -96,10 +97,39 @@ public boolean deleteTicket(final String ticketId) {
         if (ticketId == null) {
             return false;
         }
-        logger.debug("Removing ticket [{}] from registry", ticketId);
+
+        final Ticket ticket = getTicket(ticketId);
+        if (ticket == null) {
+            return false;
+        }
+
+        if (ticket instanceof TicketGrantingTicket) {
+            logger.debug("Removing ticket [{}] and its children from the registry.", ticket);
+            return deleteTicketAndChildren((TicketGrantingTicket) ticket);
+        }
+
+        logger.debug("Removing ticket [{}] from the registry.", ticket);
         return (this.cache.remove(ticketId) != null);
     }
 
+    /**
+     * Delete the TGT and all of its service tickets.
+     *
+     * @param ticket the ticket
+     * @return boolean indicating wether ticket was deleted or not
+     */
+    private boolean deleteTicketAndChildren(final TicketGrantingTicket ticket) {
+        // delete service tickets
+        final Map<String, Service> services = ticket.getServices();
+        if (services != null) {
+            for (final Map.Entry<String, Service> entry : services.entrySet()) {
+                this.cache.remove(entry.getKey());
```

Comment:
Please add a log statement here to note that the ST is being deleted. It's possible that the entry is removed from the cache by GC, so if you are not able to delete the entry, that should be logged too.